File: web/src/types/task.ts
// 任务相关类型定义（与后端 /api/v1/tasks 字段命名保持一致：snake_case）

export interface Task {
  id: string;
  title: string;
  description?: string;
  task_type: string;
  status: 'pending' | 'assigned' | 'in_progress' | 'completed' | 'cancelled';
  priority: 'low' | 'medium' | 'high' | 'urgent';
  created_by?: {
    id: string;
    username: string;
    email: string;
  };
  assigned_to?: {
    id: string;
    username: string;
    email: string;
  };
  assigned_at?: string;
  related_object_type?: string;
  related_object_id?: string;
  task_data?: unknown;
  due_date?: string;
  completed_at?: string;
  result?: any;
  notes?: string;
  created_at: string;
  updated_at: string;
}

export interface CreateTaskRequest {
  title: string;
  description?: string;
  task_type: string;
  priority?: 'low' | 'medium' | 'high' | 'urgent';
  due_date?: string;
  related_object_type?: string;
  related_object_id?: string;
  task_data?: unknown;
  assigned_to?: string;
}

export interface UpdateTaskRequest {
  title?: string;
  description?: string;
  status?: 'pending' | 'assigned' | 'in_progress' | 'completed' | 'cancelled';
  priority?: 'low' | 'medium' | 'high' | 'urgent';
  notes?: string;
  result?: any;
}

export interface TaskFilters {
  status?: string;
  priority?: string;
  task_type?: string;
  assigned_to?: string;
  created_by?: string;
  search?: string;
  date_range?: {
    start?: string;
    end?: string;
  };
}

export interface TaskStats {
  total: number;
  pending: number;
  assigned: number;
  inProgress: number;
  completed: number;
  cancelled: number;
  urgent: number;
}

// 任务类型标签映射
export const TASK_TYPE_LABELS: Record<string, string> = {
  new_user_reception: '新用户接待',
  consultation_upgrade: '咨询升级',
  system_exception: '系统异常',
  periodic_followup: '定期回访',
  prescription_review: '处方审核',
  medical_consultation: '医疗咨询',
  system_maintenance: '系统维护',
  user_feedback: '用户反馈',
  // MVP：可治理任务中枢（制造业销售视角）
  create_ticket: '创建工单',
  assign: '分派',
  transfer: '转派',
  set_sla: '设置SLA/提醒',
  create_followup: '创建跟进任务',
  sensitive_guard: '敏感拦截（待确认）',
};

// 任务状态标签映射
export const TASK_STATUS_LABELS: Record<string, string> = {
  pending: '待认领',
  assigned: '已分配',
  in_progress: '进行中',
  completed: '已完成',
  cancelled: '已取消'
};

// 任务优先级标签映射
export const TASK_PRIORITY_LABELS: Record<string, string> = {
  low: '低',
  medium: '中',
  high: '高',
  urgent: '紧急'
};

File: api/app/tasks/schemas/task.py
"""
待办任务相关的Pydantic模型
"""
from datetime import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field


class TaskBase(BaseModel):
    """待办任务基础模型"""
    title: str = Field(..., min_length=1, max_length=500, description="任务标题")
    description: Optional[str] = Field(None, max_length=2000, description="任务描述")
    task_type: str = Field(..., description="任务类型")
    priority: str = Field("medium", description="任务优先级：low, medium, high, urgent")
    due_date: Optional[datetime] = Field(None, description="截止时间")


class CreateTaskRequest(TaskBase):
    """创建待办任务请求"""
    related_object_type: Optional[str] = Field(None, description="关联对象类型")
    related_object_id: Optional[str] = Field(None, description="关联对象ID")
    task_data: Optional[Dict[str, Any]] = Field(None, description="任务相关数据")
    assigned_to: Optional[str] = Field(None, description="分配给用户ID")


class UpdateTaskRequest(BaseModel):
    """更新待办任务请求"""
    title: Optional[str] = Field(None, min_length=1, max_length=500, description="任务标题")
    description: Optional[str] = Field(None, max_length=2000, description="任务描述")
    status: Optional[str] = Field(None, description="任务状态")
    priority: Optional[str] = Field(None, description="任务优先级")
    notes: Optional[str] = Field(None, description="处理备注")
    result: Optional[Dict[str, Any]] = Field(None, description="处理结果")


class UserInfo(BaseModel):
    """用户信息"""
    id: str
    username: str
    email: str


class TaskResponse(TaskBase):
    """待办任务响应模型"""
    id: str
    status: str
    created_by: Optional[UserInfo] = None
    assigned_to: Optional[UserInfo] = None
    assigned_at: Optional[datetime] = None
    related_object_type: Optional[str] = None
    related_object_id: Optional[str] = None
    task_data: Optional[Dict[str, Any]] = None
    completed_at: Optional[datetime] = None
    result: Optional[Dict[str, Any]] = None
    notes: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    @staticmethod
    def from_model(task) -> "TaskResponse":
        """从待办任务模型转换为响应模型"""
        return TaskResponse(
            id=task.id,
            title=task.title,
            description=task.description,
            task_type=task.task_type,
            status=task.status,
            priority=task.priority,
            due_date=task.due_date,
            created_by=UserInfo(
                id=task.created_by_user.id,
                username=task.created_by_user.username,
                email=task.created_by_user.email
            ) if task.created_by_user else None,
            assigned_to=UserInfo(
                id=task.assigned_to_user.id,
                username=task.assigned_to_user.username,
                email=task.assigned_to_user.email
            ) if task.assigned_to_user else None,
            assigned_at=task.assigned_at,
            related_object_type=task.related_object_type,
            related_object_id=task.related_object_id,
            task_data=task.task_data,
            completed_at=task.completed_at,
            result=task.result,
            notes=task.notes,
            created_at=task.created_at,
            updated_at=task.updated_at
        )


class ClaimTaskRequest(BaseModel):
    """认领任务请求"""
    pass  # 不需要额外参数，用户ID从认证中获取
